sessiontxn/legacy/provider.go

```diff
@@ -190,14 +191,33 @@ func (p *SimpleTxnContextProvider) OnStmtRetry(_ context.Context) error {
 }
 
 // Advise is used to give advice to provider
-func (p *SimpleTxnContextProvider) Advise(tp sessiontxn.AdviceType) error {
+func (p *SimpleTxnContextProvider) Advise(tp sessiontxn.AdviceType, _ []any) error {
 	switch tp {
 	case sessiontxn.AdviceWarmUp:
-		p.Sctx.PrepareTSFuture(p.Ctx)
+		return p.prepareTSFuture()
 	}
 	return nil
 }
 
+func (p *SimpleTxnContextProvider) prepareTSFuture() error {
+	if p.Sctx.GetSessionVars().SnapshotTS != 0 || staleread.IsStmtStaleness(p.Sctx) || p.Sctx.GetPreparedTSFuture() != nil {
+		return nil
+	}
+
+	txn, err := p.Sctx.Txn(false)
+	if err != nil {
+		return err
+	}
+
+	if txn.Valid() {
+		return nil
+	}
+
+	txnScope := p.Sctx.GetSessionVars().CheckAndGetTxnScope()
+	future := sessiontxn.NewOracleFuture(p.Ctx, p.Sctx, txnScope)
+	return p.Sctx.PrepareTSFuture(p.Ctx, future, txnScope)
+}
+
 // activeTxn actives the txn
 func (p *SimpleTxnContextProvider) activeTxn() (kv.Transaction, error) {
```

Comment:
I suggest renaming it to `activateTxn`, if it's what the method does.

Reply:
done